core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -504,16 +510,23 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
-    private void registerBuiltinFunctionScoreParsers() {
-        registerFunctionScoreParser(new ScriptScoreFunctionParser());
-        registerFunctionScoreParser(new GaussDecayFunctionParser());
-        registerFunctionScoreParser(new LinearDecayFunctionParser());
-        registerFunctionScoreParser(new ExponentialDecayFunctionParser());
-        registerFunctionScoreParser(new RandomScoreFunctionParser());
-        registerFunctionScoreParser(new FieldValueFactorFunctionParser());
+    private void registerBuiltinScoreFunctionParsers() {
+        registerScoreFunction(ScriptScoreFunctionBuilder.FIELD, ScriptScoreFunctionBuilder::fromXContent, ScriptScoreFunctionBuilder.NAME,
```

Comment:
I'm fine with parsefield.prefferedName() name returning the NAME constant if that is also what we are using in getWritebaleName. In the original discussion this was just the first entry in an array, I think this is so much better now, thanks again. I was also going to suggest that we add tests for getWritableName().equals(QUERY/FUNCTION_NAME_FIELD.getPrefferedName()) somewhere so we check this, but we can do that as follow up.